waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -605,8 +627,10 @@
     (map
       (fn [service-id]
         (let [deployment-error (get-in service-id->deployment-error [service-id :data])
-              service (or (get service-id->service service-id) (create-empty-service service-id))]
+              service (or (get service-id->service service-id) (create-empty-service service-id))
+              k8s-event-deployment-error (get-k8s-event-deployment-error service)]
           (cond-> service
+            k8s-event-deployment-error (assoc :deployment-error k8s-event-deployment-error)
```

Comment:
Can we add a comment on the order in which the deployment errors will show up, e.g. now, failed create errors will be ignored if there are other errors. Alternatively, should we be appending to the string?

Reply:
I decided to prefer the deployment error from the cache, which currently exclusively represents errors from the k8s API in response to a replicaset create request. I would actually only expect one type of deployment error or the other to be present at any time. I'll add a comment touching on these things.